internal/kafka/internal/services/clusters.go

```diff
@@ -881,3 +889,46 @@ func (c *clusterService) FindStreamingUnitCountByClusterAndInstanceType() (Kafka
 
 	return streamingUnitsCountPerCluster, nil
 }
+
+type ClusterSizeCountPerInstanceType struct {
+	SizeId       string
+	Count        int64
+	InstanceType string
+}
+
+func (c *clusterService) ComputeConsumedStreamingUnitCountPerInstanceType(clusterID string) (StreamingUnitCountPerInstanceType, error) {
+	dbConn := c.connectionFactory.New()
+	var sizeCountsPerInstanceType []*ClusterSizeCountPerInstanceType
+	if err := dbConn.Model(&dbapi.KafkaRequest{}).
+		Select("size_id, instance_type, count(1) as Count").
+		Group("size_id, instance_type").
+		Where("cluster_id = ?", clusterID).
+		Where("status not in (?)", kafkaStatusesThatNoLongerConsumeResourcesInTheDataPlane).
+		Scan(&sizeCountsPerInstanceType).Error; err != nil {
+		return nil, apiErrors.NewWithCause(apiErrors.ErrorGeneral, err, "failed to get count of sizes of a cluster")
+	}
+
+	streamingUnitCountPerInstanceType := StreamingUnitCountPerInstanceType{
+		kafkaTypes.DEVELOPER: 0,
+		kafkaTypes.STANDARD:  0,
+	}
+
+	for _, sizeCount := range sizeCountsPerInstanceType {
+		instSize, err := c.kafkaConfig.GetKafkaInstanceSize(sizeCount.InstanceType, sizeCount.SizeId)
+		if err != nil {
+			return nil, err
+		}
+
+		streamingUnitCount := int64(instSize.CapacityConsumed) * sizeCount.Count
+		instanceType := kafkaTypes.KafkaInstanceType(sizeCount.InstanceType)
+		existingCount, ok := streamingUnitCountPerInstanceType[instanceType]
+
+		if !ok {
```

Comment:
The map is already initialized with the instance types and a count of zero. We could shorten this to just:
```
streamingUnitCountPerInstanceType[instanceType] += streamingUnitCount
```